emacs/tern.el

```diff
@@ -510,24 +512,28 @@ list of strings, giving the binary name and arguments.")
 
 ;; Mode plumbing
 
-(defun tern-before-change (start end)
+(defun tern-after-change (start end prev-length)
+  "Track the dirty area of the buffer."
   (if tern-buffer-is-dirty
-      (setf (car tern-buffer-is-dirty) (min (car tern-buffer-is-dirty) start)
-            (cdr tern-buffer-is-dirty) (max (cdr tern-buffer-is-dirty) end))
-    (setf tern-buffer-is-dirty (cons start end)))
-  (when (> (- (cdr tern-buffer-is-dirty) (car tern-buffer-is-dirty)) 4000)
-    (run-at-time "200 millisec" nil
-                 (lambda (buf)
-                   (with-current-buffer buf
-                     (when tern-buffer-is-dirty
-                       (setf tern-buffer-is-dirty nil)
-                       (tern-send-buffer-to-server))))
-                 (current-buffer)))
-  (setf tern-last-point-pos nil)
-  (when (and tern-last-argument-hints (<= start (car tern-last-argument-hints)))
-    (setf tern-last-argument-hints nil))
-  (when (and tern-last-completions (<= start (cadr tern-last-completions)))
-    (setf tern-last-completions nil)))
+      (setf tern-buffer-is-dirty (cons (min start (car tern-buffer-is-dirty))
+                                       (max end   (cdr tern-buffer-is-dirty))))
+    (setf tern-buffer-is-dirty (cons start end))))
+
+(defvar tern-idle-time 1
```

Comment:
That is 1 second, right? Why so low? Even during regular typing, you might get a one-second pause, and since Tern might do something mildly expensive, I'd be tempted to set this higher (5 seconds?)